### PR TITLE
 Use new form-/twig-extensions 

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this will break a lot.

## Changelog

```markdown
### Changed
- Replaced `sonata-project/core-bundle` classes with `form-extensions` and `twig-extensions`

### Removed
- Removed `sonata-project/core-bundle` dependency
```

### CODE REVIEW

**Comment by core23** [comment]
Ready to review :)

File: tests/Resources/XliffTest.php [hunk 1]
```diff
@@ -13,15 +13,71 @@
...
+use Symfony\Component\Translation\Exception\InvalidResourceException;
+use Symfony\Component\Translation\Loader\XliffFileLoader;
 
-class XliffTest extends XliffValidatorTestCase
+final class XliffTest extends TestCase
 {
+    /**
+     * @var XliffFileLoader
+     */
+    protected $loader;
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
    private $loader;

```

File: tests/Resources/XliffTest.php [hunk 2]
```diff
@@ -13,15 +13,71 @@
...
+    /**
+     * @var string[]
+     */
+    protected $errors = [];
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
    private $errors = [];

```

File: tests/Resources/XliffTest.php [hunk 3]
```diff
@@ -13,15 +13,71 @@
...
+
+    public function setUp()
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
    public function setUp(): void
```

File: tests/Resources/XliffTest.php [hunk 4]
```diff
@@ -13,15 +13,71 @@
...
+    {
+        $this->loader = new XliffFileLoader();
+    }
+
+    /**
+     * @dataProvider getXliffPaths
+     */
+    public function testXliff($path)
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
    public function testXliff(string $path): void
```

File: tests/Resources/XliffTest.php [hunk 5]
```diff
@@ -13,15 +13,71 @@
...
      */
     public function getXliffPaths()
     {
         return [[__DIR__.'/../../Resources/translations']];
     }
+
+    /**
+     * @param string $file The path to the xliff file
+     */
+    protected function validateXliff($file)
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
    private function validateXliff(string $file): void
```

File: tests/Resources/XliffTest.php [hunk 6]
```diff
@@ -13,15 +13,71 @@
...
+            $this->assertTrue(true, sprintf('Successful loading file: %s', $file));
+        } catch (InvalidResourceException $e) {
+            $this->errors[] = sprintf('%s => %s', $file, $e->getMessage());
+        }
+    }
+
+    /**
+     * @param string $path The path to lookup for Xliff file
+     */
+    protected function validatePath($path)
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
    private function validatePath(string $path): void
```

File: tests/Resources/XliffTest.php [hunk 7]
```diff
@@ -13,15 +13,71 @@
...
+            0,
+            $this->errors,
+            sprintf('Unable to parse xliff files: %s', implode(', ', $this->errors))
+        );
+    }
+
     /**
      * @return array List all path to validate xliff
      */
     public function getXliffPaths()
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
     public function getXliffPaths(): array
```

File: tests/Resources/XliffTest.php [hunk 8]
```diff
@@ -13,15 +13,71 @@
...
+        }
+        $this->assertCount(
+            0,
+            $this->errors,
+            sprintf('Unable to parse xliff files: %s', implode(', ', $this->errors))
+        );
+    }
+
     /**
      * @return array List all path to validate xliff
```
Comment (greg0ire) [inline tests/Resources/XliffTest.php]:
```suggestion
      * @return string[] List all path to validate xliff
```

File: tests/Resources/XliffTest.php (hunk 3 above)
Comment (OskarStark) [reply tests/Resources/XliffTest.php]: protected?

Comment (greg0ire) [reply tests/Resources/XliffTest.php]: I edited 👍

**Comment by core23** [comment]
Depends on https://github.com/sonata-project/SonataBlockBundle/pull/567

File: UPGRADE-4.0.md [hunk 9]
```diff
@@ -1,6 +1,14 @@
 UPGRADE FROM 3.X to 4.0
 =======================
 
+## Removed CoreBundle
+
+The SonataCoreBundle dependency was removed and replaced with small libraries.
```
Comment (OskarStark) [inline UPGRADE-4.0.md]: with splitted, smaller ...

File: UPGRADE-4.0.md [hunk 10]
```diff
@@ -1,6 +1,14 @@
 UPGRADE FROM 3.X to 4.0
 =======================
 
+## Removed CoreBundle
+
+The SonataCoreBundle dependency was removed and replaced with splitted, smaller libraries.
```
Comment (greg0ire) [inline UPGRADE-4.0.md]:
```suggestion
The SonataCoreBundle dependency was removed and replaced with split, smaller libraries.
```

**Comment by OskarStark** [comment]
Thanks @core23 👍

**Pull request merged**